### ISSUE TITLE
wifi.recon using ALFA AWUS036NH on Linux doesn't populate clients column in wifi.show

### ISSUE THREAD

**Issue by wcrigger** [bettercap/bettercap#89]
### Description

wifi.recon does not populate the number of clients attached to APs using latest build on Linux.  Similar behavior on OSX, more info below. 

### Environment

* Linux Mint 18.1 Serena (4.4.0-53-generic amd64)
* go version devel +9745eed Fri Jan 12 05:25:55 2018 +0000 linux/amd64
* ALFA AWUS036NH USB wifi
* screen -dmS bcwlan sudo bettercap -iface wlan0 -caplet caplets/airodump.cap
* modified airodump.cap (no ticker, ignores)
```
set $ {by}{fw}{env.iface.name}{reset} {bold}» {reset}
set ticker.commands clear; wifi.show

# uncomment to disable channel hopping
# set wifi.recon.channel 1

wifi.recon on
#ticker on
events.clear
clear

events.ignore wifi.client.probe
events.ignore wifi.ap.lost
```

* Interactive session commands you are using (if any)
wifi.show

* **Full debug output** while reproducing the issue ( `bettercap -debug ...` ).
https://pastebin.com/Vfij8KqC

### Steps to Reproduce

1. Start bettercap with: sudo bettercap -iface wlan0 -caplet caplets/airodump.cap (modified airodump.cap above)
2. Wait (in example, I did not, but I have given it about an hour with same, no clients, result)
3. wifi.show

**Expected behavior:** 
I am expecting to see the clients column in wifi.show populate. 

**Actual behavior:** 
On Linux, with monitor mode enabled prior to bettercap being started, I see no clients on any APs in wifi.show.  As a result, wifi.deauth isn't much fun =P  The only time I've seen client populate was on OSX when I was already connected to an AP when bettercap/wifi went to monitoring mode...clients populated for my AP, but not for other nearby APs.

**Comment by evilsocket** [comment]
Is there any traffic going from/to those clients?

**Comment by wcrigger** [comment]
Well, yes.....I can't speak to all of the APs....but for sure on the lngw ones (mine), I have many 802.11 devices active on the APs.  I'm wondering if it has something to do with how I'm setting up the interface prior to executing bettercap:

```
sudo rfkill unblock wifi
sudo rfkill unblock all
sudo rfkill list
sudo iwconfig wlan0 mode Monitor
sudo ifconfig wlan0 up
```

This was my workaround to get channel hopping working correctly....without doing this, i was getting an error - device busy every 250ms when it'd try to change channels.  I believe that was because NetworkManager (boo) was connecting to my AP in managed mode and kind of locked on the channel.  I think.

Anyway, I can generate substantial traffic on an AP, disconnect/reconnect, etc, or whatever type of troubleshooting you'd like me to do.

Thanks!

**Comment by wcrigger** [comment]
I turned on promisc mode on wlan0, however, that didn't seem to have any result.  Doubt that does much considering wlan0 is in Monitor mode per iwconfig wlan0.  I also disconnected and reconnected a couple of times on a couple of devices, generated significant, although short spurts of traffic to see if that made a difference......and unfortunately, not that I can tell.

**Comment by evilsocket** [comment]
you don't have to do anything rather than make sure nothing is interfering with the device ( NetworkManager, etc ), then just start bettercap ... as for client traffic, try from one of the clients to browse web pages for a few seconds and see if the client comes up.

**Comment by wcrigger** [comment]
Yea, no bueno....for the past several minutes I've been browsing utilizing one of the APs listed, but unfortunately no clients populated still.  IMO the sent traffic is on the low end as well (most under 1mb), and its worth noting that the received (RECVD) column is also not populated for any.  sad face.

**Comment by wcrigger** [comment]
this ALFA card also may be showing its age.

**Comment by evilsocket** [comment]
it happened to me with some alfas that for some reason they were not getting any data frames, then i unplugged and plugged it back and it worked ... did u try? :D

**Comment by wcrigger** [comment]
lol...it's physically in a different part of  the house, but let me give that a shot...also the traffic I have been sending was 802.11n and all of the channels reported in wifi.show are of the 2.4Ghz frequency.  Hmm, yea, still have a couple more things I can try.

**Comment by evilsocket** [comment]
that sounds too close to the issue i had not to be considered a hardware problem ...

**Comment by wcrigger** [comment]
OK, power cycled it, restarted bettercap, same results.

it works just fine in managed mode connected to an AP.  Its up high and the radio is good, so it picks up APs throughout the neighborhood.

Note that the internal wifi card on my MBP does the same thing for all other APs that I'm not connected to when I start bettercap.  It will list the clients on my AP around the time it puts the interface in monitor mode, but no other clients on other networks ever show up.

When you referred to not having anything interfering with the device, does that apply to OSX as well?  Like, if I'm utilizing wifi (it is a laptop after all) when I start bettercap, should that negatively affect it in any way?  Would it be worth connecting the laptop to ethernet, and disabling or disconnecting the WLAN prior to bettercap execution?

Maybe I should have asked this first, but is it expected for bettercap to pick up clients on APs you have not authenticated against?

Thanks again!

**Comment by evilsocket** [comment]
Yes it is expected, but as I said if bettercap is able to pick management frames but not data frames, that's a hardware problem usually, if I'll manage to reproduce I'll reopen this issue of course.

**Comment by wcrigger** [comment]
OK, thanks.  I'm buying a 2nd ALFA and will see how that turns out.  Also, will attempt to get clients populated for unauthenticated APs on my MBP.....if that doesn't work should I submit another issue (for different platform - this subject and details surrounds linux)?

**Comment by evilsocket** [comment]
Please, unless you're sure your hardware is expected to work in monitor and injection mode, do **not** submit any issue.
